carma_wm/include/carma_wm/WMTestLibForGuidance.h

```diff
@@ -0,0 +1,458 @@
+
+    auto predNearestLanelet = cmw->getMap()->laneletLayer.nearest(object_center_pred, 1)[0]; 
+    if (!boost::geometry::within(object_center_pred, predNearestLanelet.polygon2d()))
+    {
+      std::cerr << "Given pred trackpos from given lanelet id does land on any lanelet in the map" << std::endl;
+    }
+    rwo.predicted_lanelet_ids.emplace_back(predNearestLanelet.id());
+    auto tp_pred = carma_wm::geometry::trackPos(predNearestLanelet,object_center_pred);
+    rwo.predicted_cross_tracks.emplace_back(tp_pred.crosstrack);
+    rwo.predicted_down_tracks.emplace_back(tp_pred.downtrack);
+
+    pred_states.push_back(ps);
+  }
+  // add time intervals
+  rwo.object.predictions = pred_states;
+  std::vector<cav_msgs::RoadwayObstacle> rw_objs = cmw->getRoadwayObjects();	
+
+  rw_objs.push_back(rwo);	
+
+  cmw->setRoadwayObjects(rw_objs);	
+}
+/**
+ * \brief Sets all lanelets in the map the given speed limit
+ *
+ * \param speed_limit speed limit value to set
+ * \param cmw CARMAWorldModel
+ *
+ * \throw std::invalid_argument if the map is not set, contains no lanelets
+ * NOTE: this overwrites existing speed limit regulatory elements in the map
+ */
+inline void setSpeedLimit (lanelet::Velocity speed_limit, std::shared_ptr<carma_wm::CARMAWorldModel> cmw)
+{
+  if (!cmw->getMap() || cmw->getMap()->laneletLayer.size() == 0)
+  {
+    throw std::invalid_argument("setSpeedLimit: Map is not set or does not contain lanelets");
+  }
+  // remove all speed limit regulatory element and set the new speed limit
+  for (auto llt : cmw->getMutableMap()->laneletLayer)
+  {
+    for (auto regem : llt.regulatoryElements())
+    {
+      if (regem->attribute(lanelet::AttributeName::Subtype).value().compare(lanelet::DigitalSpeedLimit::RuleName) == 0)
+      {
+        cmw->getMutableMap()->remove(llt, regem);
+      }
+    }
+    lanelet::DigitalSpeedLimitPtr sl = std::make_shared<lanelet::DigitalSpeedLimit>(lanelet::DigitalSpeedLimit::buildData(lanelet::utils::getId(), speed_limit, {llt}, {},
+                                                     { lanelet::Participants::VehicleCar }));
+    cmw->getMutableMap()->update(llt, sl);
+  }
+  std::cout << "Set the new speed limit!" << std::endl;
+}
+
+/**
+ * \brief Sets the route by series of lanelets. 
+ *
+ * \param lanelets lanelets that you want the route to follow
+ * \param cmw CARMAWorldModel shared_ptr
+ *
+ * \throw std::invalid_argument if fewer than 2 lanelets passed, or the given lanelets are not routable
+ */
+inline void setRouteByLanelets (std::vector<lanelet::Lanelet> lanelets, std::shared_ptr<carma_wm::CARMAWorldModel> cmw)
+{
+  // Build routing graph from map	
+  //auto map_graph = cmw->getMapRoutingGraph();
+  lanelet::traffic_rules::TrafficRulesUPtr traffic_rules = lanelet::traffic_rules::TrafficRulesFactory::create(
+      lanelet::Locations::Germany, lanelet::Participants::VehicleCar);
```

Comment:
Use the CarmaUsTrafficRules here. You should be able to access it just by calling cmw->getTrafficRules().